File: EstructuraDeDatos/Python/ModulosPY/CAleatorios.py
# CAleatorios.py
import time
import random
t = int(time.time())
random.seed(t)
class CAleatorios:
    def __init__(self, a, n):
        self.a = a
        self.n = n
    def Aleatorio(elm, m, n):
        return random.randint(m, n)
    def GetVal(elm, i):
        return elm.a[i]

    def Buscar(elm, tl, x):
        for pos in range(tl):
            if x == elm.a[pos]:
                return -1
    def Generar(elm):
        i = 0
        n = elm.n
        while (i < n):
            alt = elm.Aleatorio(0, n - 1)
            pos =elm.Buscar(i, alt)
            if pos<0:
                elm.a.append(alt)
                i += 1
                
    def Mostrar(elm):
        cad = ""
        for i in range(elm.n):
            cad += str(elm.a[i])+ " "
        print(cad)

File: EstructuraDeDatos/CAleatorios.cpp
#include "CAleatorios.h"
CAleatorios::CAleatorios(int n)
{
 //ctor
 int pos, alt;
 srand((unsigned)time(NULL));
 if (n == 0)
 return;
 this->n = n;
 a = new int[n];
 for (int i = 0;;) {
 alt = Aleatorio(0, n - 1);
 pos = Buscar(i, alt);
 if (pos < 0) {
 a[i] = alt;
 i ++;
 }
 if (i >= n) break;

  }
}
CAleatorios::~CAleatorios()
{
 //dtor
 delete a;
}
void CAleatorios::Mostrar()
{
 for (int i = 0; i < n; i ++)
 cout << a[i] << " ";
 cout << endl;
}
unsigned CAleatorios::Aleatorio(unsigned A, unsigned B)
{
 return (A + abs(rand())%(B-A+1));
}
int CAleatorios::Buscar(int tl, int x)
{
 //Buscar
 int pos;
 for (pos = 0; pos < tl; pos ++)
 if (x == a[pos])
 return pos;
 return -1;
}
